inventory_management_system_api/cli/create.py

```diff
@@ -30,8 +48,137 @@ def system_type():
+        console.print("Below is a list of the current system types available:")
+        display_indexed_system_types(system_types)
+        _, src_system_type = ask_user_for_index_selection(
+            "Please enter the index of the [green]src_system_type[/]", system_types
+        )
+
+    return src_system_type, dst_system_type, dst_usage_status
+
+
+def display_user_constructed_rule(
+    rule_type: RuleType, src_system_type: SystemTypeOut, dst_system_type: SystemTypeOut, dst_usage_status: SystemTypeOut
+):
+    """Displays a user constructed rule obtained from `get_user_constructed_rule`."""
+
+    # Output the selected rule
+    console.print(f"You are creating the following {rule_type} rule:")
+    table = Table("src_system_type_id", "dst_system_type_id", "dst_usage_status_id")
+    table.add_row(
+        (f"{src_system_type.id} [orange1]({src_system_type.value})[/]" if src_system_type else "None"),
+        (f"{dst_system_type.id} [orange1]({dst_system_type.value})[/]" if dst_system_type else "None"),
+        (f"{dst_usage_status.id} [orange1]({dst_usage_status.value})[/]" if dst_usage_status else "None"),
+    )
+    console.print(table)
+    console.print()
+
+    # Output an explanation of the rule
+    display_rule_explanation(src_system_type, dst_system_type, dst_usage_status)
+
+
+@app.command()
+def rule(rule_type: Annotated[RuleType, typer.Argument()]):
+    """Creates a rule."""
+
+    # Acquire the required services
+    database = get_database()
+    rules_service = RuleService(RuleRepo(database))
+    system_type_service = SystemTypeService(SystemTypeRepo(database))
+    usage_status_service = UsageStatusService(UsageStatusRepo(database))
+    rules_collection = database.rules
+
+    # Obtain a list of all existing rules, system types and usage statuses
+    rules = rules_service.list(None, None)
+    system_types = system_type_service.list()
+    usage_statuses = usage_status_service.list()
+
+    # Display a table of existing rules for reference
+    console.print("Below is the current list of the current rules available:")
+    display_indexed_rules(rules)
+
+    # Get the user constructed rule
+    src_system_type, dst_system_type, dst_usage_status = get_user_constructed_rule(
+        rule_type, system_types, usage_statuses
+    )
+
+    # Ensure the same rule doesn't already exist
+    rule_data = {
+        "src_system_type_id": CustomObjectId(src_system_type.id) if src_system_type else None,
+        "dst_system_type_id": CustomObjectId(dst_system_type.id) if dst_system_type else None,
+        "dst_usage_status_id": CustomObjectId(dst_usage_status.id) if dst_usage_status else None,
+    }
+    if rules_collection.find_one(rule_data):
+        exit_with_error("The selected rule already exists!")
+
+    # Output the user selected rule and request confirmation before adding it
+    display_user_constructed_rule(rule_type, src_system_type, dst_system_type, dst_usage_status)
```

Comment:
I get the following warning for `dst_usage_status`: `Expected type 'SystemTypeOut', got 'UsageStatusOut | None' instead`